plugins/bulk-import/src/utils/repository-utils.tsx

```diff
@@ -86,44 +41,63 @@ export const createData = (
   id: number,
   name: string,
   url: string,
-  catalogInfoYaml: string,
+  catalogInfoYamlStatus: string,
+  entityOwner: string,
   organization?: string,
-  selectedRepositories?: number,
 ): AddRepositoriesData => {
   return {
     id,
-    name,
-    url,
-    organization,
-    selectedRepositories,
+    repoName: name,
+    repoUrl: url,
+    organizationUrl: organization,
     catalogInfoYaml: {
-      status: catalogInfoYaml,
-      yaml: '',
+      status: catalogInfoYamlStatus,
+      prTemplate: {
+        prTitle: 'This is the pull request title',
+        prDescription: 'This is the description of the pull request',
+        componentName: name,
+        entityOwner,
+        useCodeOwnersFile: false,
+        yaml: { kind: 'Component', apiVersion: 'v1', metadata: { name } },
+      },
     },
   };
 };
 
 export const createOrganizationData = (
-  id: number,
-  name: string,
-  url: string,
   repositories: AddRepositoriesData[],
-): AddRepositoriesData => {
-  return {
-    id,
-    name,
-    url,
-    repositories,
-    selectedRepositories: 0,
-  };
+): AddRepositoriesData[] => {
+  let id = 1;
+  return repositories.reduce(
+    (acc: AddRepositoriesData[], repo: AddRepositoriesData) => {
+      const org = acc.find(a => a.organizationUrl === repo.organizationUrl);
+      if (org?.repositories) {
+        org.repositories.push(repo);
+      } else {
+        acc.push({
+          id,
+          orgName: repo.organizationUrl,
+          organizationUrl: repo.organizationUrl,
+          repositories: [repo],
+          selectedRepositories: [],
+        });
+        id++;
```

Comment:
can we use some unique id here from data i.e `repo.id` or if you need index then reducer does return index as well.

Reply:
yes, using `repo.id` now